src/coreComponents/linearAlgebra/interfaces/hypre/HypreExport.cpp

```diff
@@ -165,6 +168,25 @@ void HypreExport::exportCRS( HypreMatrix const & mat,
   GEOSX_LAI_CHECK_ERROR( hypre_CSRMatrixDestroy( localMatrix ) );
 }
 
+namespace hypre
+{
+
+hypre_Vector * parVectorToVectorAll( hypre_ParVector * const vec )
+{
+  if( hypre_ParVectorMemoryLocation( vec ) == HYPRE_MEMORY_HOST )
+  {
+    return hypre_ParVectorToVectorAll( vec );
+  }
+
+  // Create a host-based clone
+  hypre_ParVector * const hostVec = hypre_ParVectorCloneDeep_v2( vec, HYPRE_MEMORY_HOST );
+  hypre_Vector * const fullVec = hypre_ParVectorToVectorAll( hostVec );
+  hypre_ParVectorDestroy( hostVec );
+  return fullVec;
+}
```

Comment:
`hypre_ParVectorToVectorAll` is not GPU-adapted and relied on vector memory being UM. Now that vector memory is device-only, the workaround is to explicitly copy everything to host before gathering.